source/extensions/omni.isaac.lab/omni/isaac/lab/envs/mdp/events.py

```diff
@@ -263,55 +263,49 @@ def randomize_actuator_gains(
-        stiffness = asset.data.default_joint_stiffness.to(asset.device).clone()
-        stiffness = _randomize_prop_by_op(
-            stiffness, stiffness_distribution_params, env_ids, joint_ids, operation=operation, distribution=distribution
-        )[env_ids][:, joint_ids]
-        asset.write_joint_stiffness_to_sim(stiffness, joint_ids=joint_ids, env_ids=env_ids)
-    # -- damping
-    if damping_distribution_params is not None:
-        damping = asset.data.default_joint_damping.to(asset.device).clone()
-        damping = _randomize_prop_by_op(
-            damping, damping_distribution_params, env_ids, joint_ids, operation=operation, distribution=distribution
-        )[env_ids][:, joint_ids]
-        asset.write_joint_damping_to_sim(damping, joint_ids=joint_ids, env_ids=env_ids)
+    def randomize(data: torch.Tensor, params: tuple[float, float]) -> torch.Tensor:
+        return _randomize_prop_by_op(
+            data, params, env_ids, target_joint_indices, operation=operation, distribution=distribution
+        )
 
+    # Loop through actuators and randomize gains
+    for actuator in asset.actuators.values():
+        # Resolve joint indices for each actuator
+        # Only target the joints specified by the asset_cfg.joint_ids (joint_ids_list)
+        if isinstance(actuator.joint_indices, slice):  # All joints of the articulation are this actuator type
+            target_joint_indices = torch.tensor(joint_ids_list, dtype=torch.int, device=asset.device)
+        else:
+            target_joint_indices = torch.tensor(
+                list(set(actuator.joint_indices).intersection(joint_ids_list)), dtype=torch.int, device=asset.device
+            )
+        if not target_joint_indices.nbytes: # Skip when there are no joints
+            continue
+        
+        all_envs = torch.arange(env.scene.num_envs, device=asset.device)
+        # Randomize stiffness
+        if stiffness_distribution_params is not None:
+            actuator.stiffness = asset.data.default_joint_stiffness.to(asset.device).clone()
+            randomize(actuator.stiffness, stiffness_distribution_params)
+            actuator.stiffness = actuator.stiffness[all_envs[:, None], actuator.joint_indices]
+            if isinstance(actuator, ImplicitActuator):
+                asset.write_joint_stiffness_to_sim(actuator.stiffness[env_ids][:, target_joint_indices], joint_ids=target_joint_indices, env_ids=env_ids)
+        # Randomize damping
+        if damping_distribution_params is not None:
+            actuator.damping = asset.data.default_joint_damping.to(asset.device).clone()
+            randomize(actuator.damping, damping_distribution_params)
+            actuator.damping = actuator.damping[all_envs[:, None], actuator.joint_indices]
+            if isinstance(actuator, ImplicitActuator):
+                asset.write_joint_damping_to_sim(actuator.damping[env_ids][:, target_joint_indices], joint_ids=target_joint_indices, env_ids=env_ids)
```

Comment:
```suggestion
        # Randomize stiffness
        if stiffness_distribution_params is not None:
            stiffness = asset.data.default_joint_stiffness.to(asset.device).clone()
            randomize(stiffness, stiffness_distribution_params)
            actuator.stiffness = stiffness[all_envs[:, None], actuator.joint_indices]
            if isinstance(actuator, ImplicitActuator):
                asset.write_joint_stiffness_to_sim(actuator.stiffness[env_ids[:, None], target_joint_indices], joint_ids=target_joint_indices, env_ids=env_ids)
        # Randomize damping
        if damping_distribution_params is not None:
            damping = asset.data.default_joint_damping.to(asset.device).clone()
            randomize(damping, damping_distribution_params)
            actuator.damping = damping[all_envs[:, None], actuator.joint_indices]
            if isinstance(actuator, ImplicitActuator):
                asset.write_joint_damping_to_sim(actuator.damping[env_ids[:, None], target_joint_indices], joint_ids=target_joint_indices, env_ids=env_ids)
```

Reply:
@MoreTore I just noticed before, I would modify the `actuator.stiffness` and `actuator.damping` values for all the envs which is not what we want. This suggestion should fix it.